tools/pubsub2inbox/output/mail.py

```diff
@@ -0,0 +1,380 @@
+                                          in self.output_config else None))
+            group_credentials = Credentials(
+                self.get_token_for_scopes([
+                    'https://www.googleapis.com/auth/cloud-identity.groups.readonly'
+                ],
+                                          service_account=self.
+                                          output_config['serviceAccountEmail']
+                                          if 'serviceAccountEmail'
+                                          in self.output_config else None))
+
+            user_service = discovery.build('admin',
+                                           'directory_v1',
+                                           credentials=user_credentials)
+            group_service = discovery.build('cloudidentity',
+                                            'v1beta1',
+                                            credentials=group_credentials)
+            new_emails = []
+            for e in to_emails:
+                request = group_service.groups().lookup()
+                request.uri += "&groupKey.id=" + e[1]
+                try:
+                    response = request.execute()
+                except errors.HttpError as exc:
+                    if exc.resp.status == 404 or exc.resp.status == 403:
+                        self.logger.debug(
+                            'Did not find group %s in Cloud Identity.' % (e[1]),
+                            extra={'response': exc.resp})
+                        response = None
+                    else:
+                        raise exc
+                if response and 'name' in response:
+                    m_request = group_service.groups().memberships().list(
+                        parent=response['name'])
+                    m_response = m_request.execute()
+                    if 'memberships' in m_response:  # If this field doesn't exist, it's probably an empty group
+                        for membership in m_response['memberships']:
+                            new_emails.append(
+                                email.utils.formataddr(
+                                    ('', membership['memberKey']['id'])))
+                else:
+                    try:
+                        u_response = user_service.users().get(
+                            userKey=e[1]).execute()
+                        if u_response:
+                            new_emails.append(e[1])
+                    except errors.HttpError as exc:
+                        if not 'ignoreNonexistentGroups' in self.output_config or not self.output_config[
+                                'ignoreNonexistentGroups']:
+                            raise Exception(
+                                'Failed to find group %s in Cloud Identity!' %
+                                e[1])
+                        else:
+                            if 'ignoreNonexistentGroups' in self.output_config and isinstance(
```

Comment:
`elif`? If you `elif` you may save one level of indentation and have more clear code.